### ISSUE TITLE
500 error on first load

### ISSUE THREAD

**Issue by Sonny-CP** [husseycoding/ajaxcartpopup#52]
I have a problem with this extension.

When a customer visits our website for the first time and they add a product to the cart the ajax popup appears with only the loading gif. It remains loading forever. 

When you refresh the page by F5 or going back it all works fine. 

When this problem occurs i get a 500 internal server error for: 

> 0aa7bba19ae36100c39c0c9c2f7acf94.js:1530 
> POST /checkout/cart/add/uenc/aHR0cHM6Ly93d3cuY2l0eXBhcnRzLm5sL2J1ZGR5c2VhdC16YWRlbC1jb21mb3J0LWdlbC1waWFnZ2lvLW1wMy10b3VyaW5nLWJydWluLW9yaWdpbmVlbC02NzUwMTlkbW0uaHRtbA,,/product/28205/form_key/riyhO2P0M8cjncYm/ 500 (Internal Server Error)

The problem is caused by:
```
request: function(url) {
    this.url = url;
    this.method = this.options.method;
    var params = Object.isString(this.options.parameters) ?
          this.options.parameters :
          Object.toQueryString(this.options.parameters);

    if (!['get', 'post'].include(this.method)) {
      params += (params ? '&' : '') + "_method=" + this.method;
      this.method = 'post';
    }

    if (params && this.method === 'get') {
      this.url += (this.url.include('?') ? '&' : '?') + params;
    }

    this.parameters = params.toQueryParams();

    try {
      var response = new Ajax.Response(this);
      if (this.options.onCreate) this.options.onCreate(response);
      Ajax.Responders.dispatch('onCreate', this, response);

      this.transport.open(this.method.toUpperCase(), this.url,
        this.options.asynchronous);

      if (this.options.asynchronous) this.respondToReadyState.bind(this).defer(1);

      this.transport.onreadystatechange = this.onStateChange.bind(this);
      this.setRequestHeaders();

      this.body = this.method == 'post' ? (this.options.postBody || params) : null;
      ->>>this.transport.send(this.body);<<<-

```

When this occurs there is also uncaught type error:
```
Uncaught TypeError: Cannot read property 'observe' of undefined
    at klass.disableCartLink (0aa7bba19ae36100c39c0c9c2f7acf94.js:14198)
    at klass.afterInit (0aa7bba19ae36100c39c0c9c2f7acf94.js:13808)
    at buddyseat-zadel-comfort-gel-piaggio-mp3-touring-bruin-origineel-675019dmm.html:13991
```

As said when you refresh the page, both errors dissapear.

**Comment by husseycoding** [comment]
Hi Sonny, thanks for the report.  I'm not sure what the issue is with your store, but the JS code you added above is not from this extension.  Also I'm not sure why you are seeing _klass_.disableCartLink and _klass_.afterInit.  The JS object is instantiated as _thiscartpopup_ so I would expect you to see _thiscartpopup_.disableCartLink and _thiscartpopup_.afterInit instead, so something is going on there.  You should test the extension on a clean install to confirm functionality is correct, or at the very least disable any other extension which could be affecting functionality, including extensions which do things like merge and minify assets.

**Comment by Sonny-CP** [comment]
That disablecartlink is from ajaxcartpopus.js

``Uncaught TypeError: Cannot read property 'observe' of undefined
    at klass.disableCartLink (ajaxcartpopup.js:412)
    at klass.afterInit (ajaxcartpopup.js:22)
    at buddyseat-zadel-comfort-gel-piaggio-mp3-touring-bruin-origineel-675019dmm.html:14043``

Also the JS above is from prototype.js and the problem only occurs when i enable the ajaxcart extension.

**Comment by husseycoding** [comment]
That is correct, but it doesn't mean there is an issue with the extension - my guess is you have some conflict with another extension or JS in your install but it's impossible to say what from the above.  My suggestion is that you setup a clean install (a local development environment should be fine) and install Ajax Cart Popup.  Once you have confirmed extension functionality is correct, gradually introduce the other extensions and theme from your store until functionality breaks to narrow down where the issue lies.

**Comment by Sonny-CP** [comment]
I disabled all extensions that use ajax and while the above error dissapears i keep getting a 500 server error on first load when product is added to cart. Visiting the addtocart link directly from an incognito browsers does work without a 500 server error.

**Comment by husseycoding** [comment]
As the error relates to the add to cart URL you should also check for any other extensions which override this request and disable those as well.
